Model evaluation guide (#1916)
moj-analytical-services/splink: docs/topic_guides/evaluation/model.md

# Model Evaluation

> The parameters of a trained Splink model build up a set of rules for scoring any pairwise records comparisons. Before scoring any pairs of records there are a number of ways to check whether your model will perform as you expect.

Reviewer: Suggestion to tweak this slightly to something like the below. To me, the parameters don't so much build the rules themselves as feature in them. I see the rules more as the relationship between the gamma value and trained param partly determined by the comparisons levels, i.e. gamma=3 -> params=blah -> weight = 14, etc

```suggestion
The parameters in a trained Splink model determine the match probability (Splink score) assigned to pairwise record comparisons. Before scoring any pairs of records there are a number of ways to check whether your model will perform as you expect.
```
Apologies if this is overly pedantic!

## Look at the model parameters

> The final model is summarised in the [match weights chart](../../charts/match_weights_chart.ipynb) with each bar in the chart signifying the match weight (i.e. the amount of positive or negative evidence for a match) for each comparison level in your model.

Reviewer: This might read slightly better?

```suggestion
The final model is summarised in the [match weights chart](../../charts/match_weights_chart.ipynb) with each bar in the chart signifying the match weight (i.e. the amount of evidence for or against a match) for each comparison level in your model.
```

> If, after some investigation, you still can't make sense of some of the match weights, take a look at the corresponding $m$ and $u$ values generated to see if they make logical sense. These can be produced with the [m u parameters chart](https://moj-analytical-services.github.io/splink/charts/m_u_parameters_chart.html).

Reviewer: Some small suggestions

```suggestion
If, after some investigation, you still can't make sense of some of the match weights, take a look at the corresponding $m$ and $u$ values generated to see if they themselves make sense. These can be viewed in the [m u parameters chart](https://moj-analytical-services.github.io/splink/charts/m_u_parameters_chart.html).
```

!!! info ""

    Remember that $Match Weight = \log_2 \frac{m}{u}$

## Look at the model training

> The process of training a model can offer some insight into its usefulness. The more stable a model is in the training process, the more reliable the outputs are.

Reviewer: ```suggestion
The behaviour of a model during training can offer some insight into its utility. The more stable a model is in the training process, the more reliable the outputs are.
```

> Stability of model training is seen clearly in the Expectation Maximaisation stage (for $m$ training):

Reviewer: ```suggestion
Stability of model training is seen clearly in the Expectation Maximisation stage (for $m$ training):
```

- Stability across EM training sessions can be seen through the [parameter estimates chart](https://moj-analytical-services.github.io/splink/charts/parameter_estimate_comparisons_chart.html)

- Stability within each session is indicated by the speed of convergence of the algorithm. This is shown in the terminal output during training. In general, the fewer iterations required to converge the better.

??? info "Example of convergence output"
    ![](./image/convergence.png)

## In summary

> Evaluating a trained model is not an exact science - there are no clear metrics to assess whether a model is good or bad at this stage. In most cases, applying human logic and heuristics is the best you can do to establish whether the model is sensible. Given the variety of potential use cases of Splink, there is no perfect, universal model, just models that can be tuned to produce useful outputs for a given application.

Reviewer: I think there are metrics which we can use to assess good v bad to an extent (eg the params charts) but they aren't the end of the story or definitive
```suggestion
Evaluating a trained model is not an exact science - there are no metrics which can definitively say whether a model is good or bad at this stage. In most cases, applying human logic and heuristics is the best you can do to establish whether the model is sensible. Given the variety of potential use cases of Splink, there is no perfect, universal model, just models that can be tuned to produce useful outputs for a given application.
```

> The tools within Splink are intended to help identify areas where your model may not be performaing as expected. In future versions releases we hope to automatically flag where there are areas of a model that require further investigation to make this process easier for the user.

Reviewer: ```suggestion
The tools within Splink are intended to help identify areas where your model may not be performing as expected. In future versions releases we hope to automatically flag where there are areas of a model that require further investigation to make this process easier for the user.
```
